### ISSUE TITLE
abstol and reltol are ignored

### ISSUE THREAD

**Issue by sjdaines** [SciML/SimpleNonlinearSolve.jl#57]
`abstol` and `reltol` keyword arguments seem to be ignored. 

```
julia> using SimpleNonlinearSolve

julia> f(u, p) = u^4 .- 2.0
julia> prob = IntervalNonlinearProblem(f, (1.0, 2.0))

julia> sol = solve(probB, Brent())  # full precision
u: 1.1892071150027212

julia> sol = solve(prob, Brent(), maxiters=3) # maxiters works
u: 1.0666666666666667

julia> sol = solve(prob, Brent(), abstol=0.1)  # abstol is ignored and gives the maximum precision
u: 1.1892071150027212
```

More fundamental problem perhaps is that there are apparently no tests for abstol and reltol ?

**Comment by ChrisRackauckas** [comment]
Yes, some of the algorithms are not complete but err on the side of correctness before performance. Definitely some early exits can be added to many of them.

**Comment by sjdaines** [comment]
Key point here I think is that depending on the function f, this can be a correctness issue, not a performance issue:
- if f is numerically noisy (eg this is the result of some numerical solution with limited precision) then a tolerance is required to reliably find a root.
- if f is a "simple" function (eg those commonly used for testing) then of course it is just an optimisation.
